src/circuit.rs

```diff
@@ -1441,4 +1445,232 @@ mod tests {
+        } else {
+            AssetBase::random(&mut rng)
+        };
+
+        // Create spent_note
+        let (spent_note_fvk, spent_note) = {
+            let sk = SpendingKey::random(&mut rng);
+            let fvk: FullViewingKey = (&sk).into();
+            let sender_address = fvk.address_at(0u32, Scope::External);
+            let rho_old = Nullifier::dummy(&mut rng);
+            let spent_note = Note::new(
+                sender_address,
+                NoteValue::from_raw(40),
+                asset_base,
+                rho_old,
+                &mut rng,
+            );
+            (fvk, spent_note)
+        };
+
+        let (dummy_sk, fvk, scope, nf_old) = if split_flag {
+            let sk = SpendingKey::random(&mut rng);
+            let fvk: FullViewingKey = (&sk).into();
+            (
+                Some(sk),
+                fvk.clone(),
+                Scope::External,
+                spent_note.nullifier(&fvk),
+            )
+        } else {
+            (
+                None,
+                spent_note_fvk.clone(),
+                spent_note_fvk
+                    .scope_for_address(&spent_note.recipient())
+                    .unwrap(),
+                spent_note.nullifier(&spent_note_fvk),
+            )
+        };
+        let ak: SpendValidatingKey = fvk.clone().into();
+        let alpha = pallas::Scalar::random(&mut rng);
+        let rk = ak.randomize(&alpha);
+
+        let output_note = {
+            let sk = SpendingKey::random(&mut rng);
+            let fvk: FullViewingKey = (&sk).into();
+            let sender_address = fvk.address_at(0u32, Scope::External);
+
+            Note::new(
+                sender_address,
+                NoteValue::from_raw(10),
+                asset_base,
+                nf_old,
+                &mut rng,
+            )
+        };
+
+        let cmx = output_note.commitment().into();
+
+        let value = if split_flag {
+            NoteValue::zero() - output_note.value()
+        } else {
+            spent_note.value() - output_note.value()
+        };
+        let rcv = ValueCommitTrapdoor::random(&mut rng);
+        let cv_net = ValueCommitment::derive(value, rcv, asset_base);
+
+        let path = MerklePath::dummy(&mut rng);
+        let anchor = path.root(spent_note.commitment().into());
+
+        let spend_info = SpendInfo {
+            dummy_sk,
+            fvk,
+            scope,
+            note: spent_note,
+            merkle_path: path,
+            split_flag,
+        };
+
+        (
+            Circuit::from_action_context_unchecked(spend_info, output_note, alpha, rcv),
+            Instance {
+                anchor,
+                cv_net,
+                nf_old,
+                rk,
+                cmx,
+                enable_spend: true,
+                enable_output: true,
+            },
+        )
+    }
+
+    fn check_proof_of_orchard_circuit(circuit: &Circuit, instance: &Instance, should_pass: bool) {
```

Comment:
Let's put this function above `generate_circuit_instance()`

Reply:
Done